txn.go

```diff
@@ -246,10 +290,10 @@ func (tr *transactionRunner) newRunnerImpl() txnRunner {
 
 // Run is defined on Runner.
 func (tr *transactionRunner) Run(transactions TransactionSource) error {
-	for i := 0; i < nrRetries; i++ {
+	for i := 0; i < tr.nrRetries; i++ {
 		ops, err := transactions(i)
 		if err == ErrTransientFailure {
-			continue
+			goto retry
```

Comment:
Seems weird to have a `goto` here. What about this?

```go
if err == ErrTransientFailure {
    tr.pause(i)
    continue
}
```

Reply:
I didn't want to have pause() twice - thought it better to just have a call to pause(0 last thing in the loop.
But I don't mind either way really